internal/internal_workflow_testsuite.go

```diff
@@ -312,14 +312,15 @@ func newTestWorkflowEnvironmentImpl(s *WorkflowTestSuite, parentRegistry *regist
 		return nil
 	}
 
-	em := mockService.EXPECT().RecordActivityTaskHeartbeat(gomock.Any(), gomock.Any(), gomock.Any()).
-		Return(&workflowservice.RecordActivityTaskHeartbeatResponse{CancelRequested: false}, nil)
-	em.Do(func(ctx context.Context, r *workflowservice.RecordActivityTaskHeartbeatRequest, opts ...grpc.CallOption) {
-		// TODO: The following will hit a data race in the gomock code where the Do() action is executed outside
```

Comment:
I could not find any information on what this warning relates to. The tests were succeeding for me and I have uncommented this. We need the heartbeat result to know whether the activity is cancelled. I will keep an eye out to see if what this is referring to resurfaces.